okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java

```diff
@@ -273,50 +285,65 @@ public Void visit(QosException.Throttle exception) {
                 Optional<Duration> backoff = backoffStrategy.nextBackoff();
                 if (!backoff.isPresent()) {
-                    log.debug("Max number of retries exceeded, failing call");
-                    callback.onFailure(call,
-                            new IOException("Failed to complete the request due to a "
-                                    + "server-side QoS condition: 503", exception));
-                } else {
-                    log.debug("Rescheduling call after backoff",
-                            SafeArg.of("backoffMillis", backoff.get().toMillis()), exception);
-                    // Redirect to the "next" URL, whichever that may be, after backing off.
-                    Optional<HttpUrl> redirectTo = urls.redirectToNext(request().url());
-                    if (!redirectTo.isPresent()) {
-                        callback.onFailure(call, new IOException("Failed to determine valid redirect URL for base "
-                                + "URLs " + urls.getBaseUrls()));
-                    } else {
-                        Request redirectedRequest = request().newBuilder()
-                                .url(redirectTo.get())
-                                .build();
-                        scheduleExecution(() -> client.newCallWithMutableState(redirectedRequest, backoffStrategy,
-                                maxNumRelocations).enqueue(callback), backoff.get());
-                    }
+                    callback.onFailure(call, new SafeIoException(
+                            "Failed to complete the request due to QosException.Unavailable",
+                            exception,
+                            UnsafeArg.of("requestUrl", call.request().url().toString())));
+                    return null;
+                }
+
+                // Redirect to the "next" URL, whichever that may be, after backing off.
+                Optional<HttpUrl> redirectTo = urls.redirectToNext(request().url());
+                if (!redirectTo.isPresent()) {
+                    callback.onFailure(call, new SafeIoException(
+                            "Failed to determine valid redirect URL after receiving QosException.Unavailable",
+                            UnsafeArg.of("requestUrl", call.request().url().toString()),
+                            UnsafeArg.of("baseUrls", urls.getBaseUrls())));
+                    return null;
                 }
+
+                log.debug("Retrying call after receiving QosException.Unavailable",
+                        SafeArg.of("backoffMillis", backoff.get().toMillis()),
+                        exception);
+                Request redirectedRequest = request().newBuilder()
+                        .url(redirectTo.get())
+                        .build();
+                scheduleExecution(() -> client.newCallWithMutableState(redirectedRequest, backoffStrategy,
+                        maxNumRelocations).enqueue(callback), backoff.get());
```

Comment:
I know you just copied code but do you mind making the method chaining more readable?
The formatting is super weird across this whole repo sadly